types/src/crypto/asymmetric_key.rs

```diff
@@ -697,6 +697,38 @@ impl From<&SecretKey> for PublicKey {
     }
 }
 
+#[cfg(any(feature = "testing", test))]
+impl PartialEq for SecretKey {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::System, Self::System) => true,
+            (Self::Ed25519(k1), Self::Ed25519(k2)) => k1.to_bytes() == k2.to_bytes(),
+            (Self::Secp256k1(k1), Self::Secp256k1(k2)) => k1.to_bytes() == k2.to_bytes(),
+            _ => false,
+        }
+    }
+}
+#[cfg(any(feature = "testing", test))]
+impl Eq for SecretKey {}
+
+#[cfg(any(feature = "testing", test))]
+impl Ord for SecretKey {
```

Comment:
Seems a bit weird to have `Ord` implemented for a `SecretKey`. Maybe it's ok for testing but this relation cannot actually be determined for keys.

Reply:
Yeah, it's only for testing, as indicated by the feature gate. I did that because it's simpler to write the builder that way.